utility/importer/brave_importer.cc

```diff
@@ -315,3 +338,233 @@ void BraveImporter::ImportStats() {
+      payments->min_visit_time =
+        braveledger_ledger::_default_min_publisher_duration;
+  }
+
+  return true;
+}
+
+bool ParseExcludedSites(BraveLedger& ledger,
+  const base::Value& session_store_json) {
+  const base::Value* site_settings = session_store_json.FindKeyOfType(
+    "siteSettings",
+    base::Value::Type::DICTIONARY);
+  if (!site_settings) {
+    LOG(ERROR) << "No entry \"siteSettings\" found in session-store-1";
+    return false;
+  }
+
+  ledger.excluded_publishers = std::vector<std::string>();
+  bool host_pattern_included;
+
+  for (const auto& item : site_settings->DictItems()) {
+    const auto& host_pattern = item.first;
+    const auto& settings = item.second;
+
+    if (!settings.is_dict())
+      continue;
+
+    if (TryFindBoolKey(&settings, "ledgerPayments", host_pattern_included)) {
+      if (!host_pattern_included) {
+        // host pattern is in a format like: `https?://travis-ci.org`
+        // The protocol part can be removed (to get publisher key)
+        size_t protocol_index = host_pattern.find("//");
+        if (protocol_index != std::string::npos) {
+          ledger.excluded_publishers.push_back(
+            host_pattern.substr(protocol_index + 2));
+        }
+      }
+    }
+  }
+
+  return true;
+}
+
+bool ParsePinnedSites(BraveLedger& ledger,
+  const base::Value& session_store_json) {
+  const base::Value* publishers = session_store_json.FindPathOfType(
+      {"ledger", "synopsis", "publishers"},
+      base::Value::Type::DICTIONARY);
+  if (!publishers) {
+    LOG(ERROR)
+      << "\"ledger\".\"synopsis\".\"publishers\" not found in session-store-1";
+    return false;
+  }
+
+  ledger.pinned_publishers = std::map<std::string, unsigned int>();
+  int pin_percentage;
+
+  for (const auto& item : publishers->DictItems()) {
+    const auto& domain = item.first;
+    const auto& settings = item.second;
+
+    if (!settings.is_dict())
+      continue;
+
+    if (TryFindIntKey(&settings, "pinPercentage", pin_percentage)) {
+      if (pin_percentage > 0) {
+        ledger.pinned_publishers.insert(
+          std::pair<std::string, unsigned int>(domain, pin_percentage));
+      }
+    }
+  }
+
+  return true;
+}
+
+void BraveImporter::ImportLedger(bool clobber_wallet) {
+  std::unique_ptr<base::Value> session_store_json = ParseBraveStateFile(
+      "session-store-1");
+  std::unique_ptr<base::Value> ledger_state_json = ParseBraveStateFile(
+      "ledger-state.json");
+  if (!(session_store_json && ledger_state_json))
+    return;
+
+  BraveLedger ledger;
+  // TODO: when initiated from the UI, user should be prompted
+  // "Are you sure you want to restore? You'll lose your wallet"
+  // If they choose yes, then caller should set `clobber_wallet` to true
+  ledger.clobber_wallet = clobber_wallet;
+
+  if (!ParseWalletPassphrase(ledger, *session_store_json)) {
+    LOG(ERROR) << "Failed to parse wallet passphrase";
+  }
+
+  if (!ParsePaymentsPreferences(ledger, *session_store_json)) {
```

Comment:
Shouldn't we cancel the wallet import if any these parsing steps fails?

Reply:
Yes we should 😄 Fixing that now...